### ISSUE TITLE
Container export / start: Failed to run: zfs mount ... filesystem already mounted

### ISSUE THREAD

**Issue by tomee03** [lxc/lxd#8439]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 20.04
 * The output of "lxc info" or if that fails:
   * Kernel version: 5.4.0-60-generic
   * LXC version:
   * LXD version: 4.11 (from snap)
   * Storage backend in use: zfs

# Issue description

We have an automatic backup script and today we had a problem with `lxc export` in 3 from 15 running containers.
Error: Error: Create backup: Backup create: Failed to run: zfs mount default/containers/xxx1: cannot mount 'default/containers/xxx1': filesystem already mounted

So I tried `lxc stop xxx1` and `lxc start xxx1` but I have the problem that I can't start the container again. 
Output: Error: Failed preparing container for start: Failed to run: zfs mount default/containers/xxx1: cannot mount 'default/containers/xxx1': filesystem already mounted
Try `lxc info --show-log xxx1` for more info

Output (lxc info --show-log xxx1)
Name: xxx1
Location: none
Remote: unix://
Architecture: x86_64
Created: 2021/01/26 18:03 UTC
Status: Stopped
Type: container
Profiles: default

Log:

---- 

That's all. I'm really frustrated, how it's possible that it will just stop working? Can you help me? I tried all examples from your previous topics.

Please help me how I can resolve this issue. All other containers working properly.

**Comment by stgraber** [comment]
Duplicate of https://github.com/lxc/lxd-pkg-snap/issues/61

**Comment by stgraber** [comment]
The usual things to try to fix this are:
 - `grep containers/xxx1 /proc/*/mountinfo`
 - `nsenter -t PID -m umount PATH` using one of the PID and one of the mount paths from above output

If that doesn't work, you may need to unmount the overmount `/var/snap/lxd/common/shmounts` path and then repeat the previous step.

If more practical, a reboot will also clear the issue for sure.

**Comment by tomee03** [comment]
Hello, I don't understand why you close the issue, obviously, lxd has a problem. Your answer not working. And marked issue `lxc/lxd-pkg-snap#61` too.

Output from umount:
`umount: /var/snap/lxd/common/shmounts/storage-pools/default/containers/xxx1: no mount point specified.`

We have production things on lxc, I'm afraid it's wrong.

Can you please describe why this problem with mounting happened?

It's because we using `lxc export`? Or you know why this state happens?

If I unmount `/var/snap/lxd/common/shmounts` I can't disrupt running containers?

Thanks

**Comment by stgraber** [comment]
I'm closing it because it's not a LXD issue (it's a snap issue) and because we already have an issue tracking this.

Can you show 'nsenter -t PID -m cat /proc/self/mountinfo' for one of the PIDs in the initial list?

That way I can make sure that clearing shmounts is the right thing to do to unblock things.

And this has nothing to do with an export or with anything you did, it's some kind of mount table handling bug which appears to occur after a system has been up for quite a while and several round of security updates for the core, core 18 and lxd components have been applied each altering the mount table.
